src/auth/models.hpp

```diff
@@ -166,14 +192,15 @@ class User final {
   void ClearRole();
 
   Permissions GetPermissions() const;
-  FineGrainedAccessPermissions GetFineGrainedAccessPermissions() const;
+  FineGrainedAccessPermissions GetFineGrainedAccessLabelPermissions() const;
+  FineGrainedAccessPermissions GetFineGrainedAccessEdgeTypePermissions() const;
 
   const std::string &username() const;
 
   const Permissions &permissions() const;
   Permissions &permissions();
-  const FineGrainedAccessPermissions &fine_grained_access_permissions() const;
-  FineGrainedAccessPermissions &fine_grained_access_permissions();
+  const FineGrainedAccessHandler &fine_grained_access_handler() const;
+  FineGrainedAccessHandler &fine_grained_access_handler();
```

Comment:
So you want me to either have only functions that provide const access or to make them public ? I coded this class and everything that goes with it by following how it was done for Permission class.

Reply:
We didn't give it much thought here, we simply reused what was written for the permissions member, like Niko pointed out. We will remove this non-const access for fine_grained_access_handler. 

I propose leaving permissions as-is for now, we could run a clean-up task on things we spotted during development

Reply:
Is it okay to leave it as it is for now and fix this and Permission class in another task because there are some function that are used by Permission class and my classes so if I change my class I will need to change everything and I believe that is not in scope of this task?